packages/volto/src/components/theme/SlotRenderer/SlotRenderer.tsx

```diff
@@ -0,0 +1,40 @@
+import { v4 as uuid } from 'uuid';
+import config from '@plone/volto/registry';
+import { useLocation } from 'react-router-dom';
+import type { Content } from '@plone/types';
+
+/*
+<SlotRenderer name="aboveContent" content={content} route={} />
+*/
+
+const SlotRenderer = ({
+  name,
+  content,
+}: {
+  name: string;
+  content: Content;
+}) => {
+  const pathname = useLocation().pathname;
+
+  let slots = config.getSlot<{ content: Content; pathname: string }>(name, {
+    content,
+    pathname,
+  });
+
+  if (!slots) {
+    return null;
+  }
+
+  return (
+    <>
+      {slots.map((component: React.ComponentType<any>) => {
+        // Weird compilation issue, ^^ that forced to re-declare the type above
+        const id = uuid();
```

Comment:
Won't this force re-rendering all slots even if their props haven't changed? I think it would make more sense to use the name as the key.

Reply:
Good catch. I've enhanced what `getSlot` returns to be able to set the key as the name of the component.